processor/src/decorators/mod.rs

```diff
@@ -278,6 +282,62 @@ where
 
         Ok(())
     }
+
+    /// Pushes the value and depth flags of a leaf indexed by `key` on a Sparse Merkle tree with
+    /// the provided `root`.
+    ///
+    /// The Sparse Merkle tree is tiered, meaning it will have leaf depths in `{16, 32, 48, 64}`.
+    /// The depth flags define the tier on which the leaf is located.
+    ///
+    /// The operand stack is expected to be arranged as follows:
+    /// - key, 4 elements.
+    /// - root of the Sparse Merkle tree, 4 elements.
+    ///
+    /// After a successful operation, the advice stack will look as follows:
+    /// - boolean flag set to `1` if the depth is `16` or `32`.
+    /// - boolean flag set to `1` if the depth is `16` or `48`.
+    /// - value word; will be zeroed if the tree don't contain a mapped value for the key.
+    ///
+    /// # Errors
+    /// Will return an error if:
+    /// - The provided Merkle root doesn't exist on the advice provider
+    /// - If, at depth `64`, the node value is not zero, and there is no associated leaf value
+    fn inject_smtget(&mut self) -> Result<(), ExecutionError> {
+        // fetch the arguments from the operand stack
+        let key = [self.stack.get(3), self.stack.get(2), self.stack.get(1), self.stack.get(0)];
+        let root = [self.stack.get(7), self.stack.get(6), self.stack.get(5), self.stack.get(4)];
+
+        // fetch the leaf depth from the Merkle store
+        // an empty tree will return depth `0`, but we always point to the minimum tier (16)
+        const TREE_DEPTH: Felt = Felt::new(64);
+        let index = &key[3];
+        let depth = self.advice_provider.get_leaf_depth(root, &TREE_DEPTH, index)?.max(16);
+
+        // fetch the node value
+        let index = index.as_int() >> (64 - depth);
+        let index = Felt::new(index);
+        let node = self.advice_provider.get_tree_node(root, &Felt::new(depth as u64), &index)?;
+
+        // set the flags
+        let empty = EmptySubtreeRoots::empty_hashes(64);
+        let is_16_or_32 = if depth == 16 || depth == 32 { ONE } else { ZERO };
+        let is_16_or_48 = if depth == 16 || depth == 48 { ONE } else { ZERO };
+        let value = if Word::from(empty[depth as usize]) == node {
+            [ZERO; WORD_SIZE]
+        } else {
+            self.advice_provider.push_stack(AdviceSource::Map { key: node })?;
+            self.advice_provider.pop_stack_word()?
+        };
+
+        // push the values
+        self.advice_provider.push_stack(AdviceSource::Value(is_16_or_32))?;
+        self.advice_provider.push_stack(AdviceSource::Value(is_16_or_48))?;
+        self.advice_provider.push_stack(AdviceSource::Value(value[3]))?;
+        self.advice_provider.push_stack(AdviceSource::Value(value[2]))?;
+        self.advice_provider.push_stack(AdviceSource::Value(value[1]))?;
+        self.advice_provider.push_stack(AdviceSource::Value(value[0]))?;
+        Ok(())
```

Comment:
I was a bit confused by the `pop_stack_word()` on line 329 at first. Why not write this as:
```Rust
let is_16_or_32 = if depth == 16 || depth == 32 { ONE } else { ZERO };
let is_16_or_48 = if depth == 16 || depth == 48 { ONE } else { ZERO };
self.advice_provider.push_stack(AdviceSource::Value(is_16_or_32))?;
self.advice_provider.push_stack(AdviceSource::Value(is_16_or_48))?;

let empty = EmptySubtreeRoots::empty_hashes(64);
let value = if Word::from(empty[depth as usize]) == node {
    self.advice_provider.push_stack(AdviceSource::Value(ZERO))?;
    self.advice_provider.push_stack(AdviceSource::Value(ZERO))?;
    self.advice_provider.push_stack(AdviceSource::Value(ZERO))?;
    self.advice_provider.push_stack(AdviceSource::Value(ZERO))?;
} else {
    self.advice_provider.push_stack(AdviceSource::Map { key: node })?;
};
```

Reply:
True! It doesn't make sense to push, pop, push. Guess I was just checking the correct endianness of the word. Fixed